src/ecs/resources/input.rs

```diff
@@ -333,9 +333,9 @@ impl InputHandler {
     }
 
     /// Returns the value of an axis by the i32 id, if the id doesn't exist this returns None.
-    pub fn axis_value(&self, id: i32) -> Option<f32> {
+    pub fn axis_value(&self, id: &str) -> Option<f32> {
```

Comment:
The `AsRef` trait exists precisely for this reason, though. It's meant to eliminate boilerplate in case users decide to pass in a `String`, `CString`, `OsString`, or perhaps even a simple array of bytes, and more importantly, it can prevent extra heap allocations on the user's end.